Resolve #1215 -- Introduced a General Scripting system for each item in order to script passive effects

cabeca1143 opened the pull request:
* Introduced `ItemPassives` namespace
* Item spells no longer use the `Spells`  namespace, they use `ItemSpells`
* "Items" folder changed, now it is organized between "Actives" and "Passives"
* Reorganized the scripts folder so now there's a "Spells", which inside you'll find the Champions and Items folder.
* Introduced `IItemScript` Interface.
* Introduced a system that assigns every item an `ItemScript`.
* Introduced Doran's Shield and Doran's Ring Passive effects as demonstration
* Updated the Tests in order to properly reflect the changes.


Resolve #1215

cabeca1143 committed:
Introduced General Item scripting in order to script item passive effects

SomeOtherDev commented on Content/LeagueSandbox-Scripts/Spells/Items/Passives/Doran's Shield.cs:
```diff
@@ -0,0 +1,29 @@
+
+        public void OnActivate(IObjAiBase owner)
+        {
+            StatsModifier.HealthRegeneration.BaseBonus += 1.2f;
```
Is this correct? 
Wiki reads: `+6 health regen per 5 seconds`
https://web.archive.org/web/20150214051041/http://leagueoflegends.wikia.com/wiki/Doran's_Shield

cabeca1143 replied:
Yeah, 6 regen per 5 seconds translates to 1.2 per second (6/5)

SomeOtherDev reviewed (commented):
Just one comment on Doran's Shield implementation

danil179 reviewed (commented):
I think that there are new spells from the last PRs. Please check if all the spells moved to their new locations (I also don't think that you really need to move the Champions folder into a new Spells folder. I prefer not to move many files in a PR, unless there is a justification).

cabeca1143 committed:
Reorganized scripts

cabeca1143 committed:
resolved

cabeca1143 commented:
> I think that there are new spells from the last PRs. Please check if all the spells moved to their new locations (I also don't think that you really need to move the Champions folder into a new Spells folder. I prefer not to move many files in a PR, unless there is a justification).

Resolved

cabeca1143 committed:
The Item list now uses ItemId instead of Name

cabeca1143 committed:
Update Inventory.cs

Items' OnDeactivate no longer get's triggered every time you sell the item (if you have multiple of said item), only when you sell the last one

danil179 reviewed (approved):
Approved. Didn't test.
